drkafka/src/main/java/com/pinterest/doctorkafka/modules/monitor/cluster/kafka/BrokerStatsMonitor.java

```diff
@@ -0,0 +1,283 @@
+      Map<TopicPartition, Long> offsets = ReplicaStatsUtil
+          .getProcessingStartOffsets(kafkaConsumer, brokerStatsTopic, startTimestampInMillis);
+
+      kafkaConsumer.unsubscribe();
+      kafkaConsumer.assign(offsets.keySet());
+      Map<TopicPartition, Long> latestOffsets = kafkaConsumer.endOffsets(offsets.keySet());
+      KafkaUtils.closeConsumer(zkUrl);
+
+      for(TopicPartition tp : latestOffsets.keySet()){
+        Processor processor = new Processor(tp, offsets.get(tp), latestOffsets.get(tp), zkUrl, securityProtocol);
+        processors.add(processor);
+        processor.run();
+      }
+
+      for (Processor processor : processors){
+        try {
+          processor.join();
+        } catch (Exception e){
+          LOG.error("Processor failed to backfill historical data", e);
+        }
+      }
+      isBackfillComplete = true;
+      LOG.info("Finish rebuilding the replica stats");
+
+      // Backfill complete, start long-running background processor
+
+      processors.clear();
+      Processor processor = new Processor(brokerStatsTopic, zkUrl, securityProtocol, consumerConfigs);
+      processors.add(processor);
+      processor.start();
+    }
+  }
+
+  /**
+   * Process brokerstats from either a topic or a topic-partition
+   */
+  private static class Processor implements Runnable {
+    private static final long BROKER_STATS_POLL_INTERVAL_MS = 200L;
+    private static final String BROKERSTATS_CONSUMER_GROUP =
+        "operator_brokerstats_group_" + OperatorUtil.getHostname();
+
+    private Consumer<byte[], byte[]> consumer;
+    private Thread thread;
+    private volatile boolean stopped = false;
+
+    private TopicPartition topicPartition;
+    private long endOffset = -1L;
+
+    // long running processor
+    public Processor(String topic, String zkUrl, SecurityProtocol securityProtocol, Properties consumerConfigs) {
+      Properties properties = OperatorUtil.createKafkaConsumerProperties(
+          zkUrl, BROKERSTATS_CONSUMER_GROUP, securityProtocol, consumerConfigs);
+      consumer = new KafkaConsumer<>(properties);
+      consumer.subscribe(Collections.singletonList(topic));
+    }
+
+    // backfill processor
+    public Processor(TopicPartition topicPartition, long startOffset, long endOffset, String zkUrl, SecurityProtocol securityProtocol){
+      this.topicPartition = topicPartition;
+      this.endOffset = endOffset;
+
+      String brokers = KafkaUtils.getBrokers(zkUrl, securityProtocol);
+      LOG.info("ZkUrl: {}, Brokers: {}", zkUrl, brokers);
+      Properties props = new Properties();
+      props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokers);
+      props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+      props.put(ConsumerConfig.GROUP_ID_CONFIG, "doctorkafka_" + topicPartition);
+      props.put(KafkaUtils.KEY_DESERIALIZER,
+          "org.apache.kafka.common.serialization.ByteArrayDeserializer");
```

Comment:
Could we take this refactoring opportunity to declare default values like `"org.apache.kafka.common.serialization.ByteArrayDeserializer"` as constants?